timesketch/frontend-ng/src/components/Studio/SigmaRuleModification.vue

```diff
@@ -0,0 +1,257 @@
+      label="Edit Sigma rule"
+      outlined
+      :color="statusColors()"
+      autocomplete="email"
+      rows="20"
+      v-model="ruleYamlTextArea"
+      background-color="statusColors()"
+      @input="parseSigma(ruleYamlTextArea)"
+      class="editSigmaRule"
+    >
+    </v-textarea>
+
+    <div>
+      <v-btn :disabled="!isParsingSuccesful" @click="addOrUpdateRule(ruleYamlTextArea)" small depressed color="primary">
+        {{ isNewRule ? 'Create Rule' : 'Update Rule' }}
+      </v-btn>
+      <div style="width: 10px; display: inline-block"></div>
+      <v-btn color="primary" text @click="$router.back()"> Cancel </v-btn>
+      <div style="width: 10px; display: inline-block"></div>
+      <v-btn @click="deleteRule(rule_uuid)" small text color="primary" :disabled="isNewRule"
+        ><v-icon>mdi-delete</v-icon></v-btn
+      >
+    </div>
+
+    <div class="alertbox" v-if="isParsingSuccesful">
+      <h3>Preview opensearch query:</h3>
+      <pre>{{ editingRule.search_query }}</pre>
+      <!-- TODO: This should become a global class how to nicely highlight a query -->
+    </div>
+  </v-container>
+</template>
+  
+<script>
+import ApiClient from '../../utils/RestApiClient'
+import { SigmaTemplates, defaultSigmaPlaceholder } from '@/utils/SigmaRuleTemplates'
+import EventBus from '../../main'
+import _ from 'lodash'
+
+export default {
+  props: ['rule_uuid', 'sigmaRule'],
+  data() {
+    return {
+      editingRule: { ruleYaml: defaultSigmaPlaceholder }, // empty state
+      status_chip_text: 'OK',
+      status_text: '',
+      ruleYamlTextArea: {},
+      SigmaTemplates: SigmaTemplates,
+      search: '',
+      isNewRule: false,
+      isUpdatingRule: false,
+      isParsingSuccesful: false,
+    }
+  },
+  watch: {
+    rule_uuid: function (newVal, oldVal) {
+      if (newVal === 'new') {
+        this.editingRule = {
+          title: 'New Sigma Rule',
+        }
+        this.isNewRule = true
+        this.isUpdatingRule = false
+        this.status_chip_text = 'Ok'
+        this.isParsingSuccesful = true
+        this.ruleYamlTextArea = defaultSigmaPlaceholder
+        this.editingRule.rule_yaml = defaultSigmaPlaceholder
+        this.parseSigma(this.editingRule.rule_yaml)
+      } else {
+        this.getRuleByUUID(newVal)
+      }
+    },
+  },
+  loaded() {
+    this.parseSigma(this.editingRule.rule_yaml)
+  },
+  mounted() {
+    if (this.rule_uuid === 'new') {
+      this.editingRule = {
+        title: 'New Sigma Rule',
+      }
+      this.isNewRule = true
+      this.isUpdatingRule = false
+      this.status_chip_text = 'Ok'
+      this.isParsingSuccesful = true
+      this.ruleYamlTextArea = defaultSigmaPlaceholder
+      this.editingRule.rule_yaml = defaultSigmaPlaceholder
+      this.parseSigma(this.editingRule.rule_yaml)
+    } else {
+      this.getRuleByUUID(this.rule_uuid)
+    }
+  },
+  methods: {
+    selectTemplate(text) {
+      var matchingTemplate = this.SigmaTemplates.find((obj) => {
```

Comment:
What happens if `find` finds no template with the selected "text"?

Reply:
It would display "No data available" in the dropdown and nothing would be availabe to select, so the pre-selected general template would still be used

Reply:
Can you capture this in the code by returning if no template was found? (otherwise you'll be getting undefined errors)